### PR TITLE
Move duplicate conversion functions to Message Helper

### LINKED ISSUE #3448: Duplicated conversion functions in different SDL places
### Bug Report

We have lots of similar conversion functions, which are located in different SDL places (MessageHelper, RAI request, Application Manager etc.). This conversion functions are duplicates of each other in terms of conversion from string to enum and vice versa. There is no unified approach to perform this conversion, that's why in similar functions this operation is realized by different means (for example, using map, switch or SchemaItem-based convertor).

Many of this functions are present in MessageHelper as static public methods, which have corresponding mocked methods, used in unit tests. That's why in unit tests we cannot check actual values of converted enums, because now this values are hard-coded in expectations of mocked methods. Also presence of duplicated methods increases possibility of errors and makes code less clear and consistent.

##### OS & Version Information
* OS/Version: Ubuntu 18
* SDL Core Version: 6.1

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #3448 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Affected unit tests are updated

### Summary
We have lots of similar conversion functions, which are located in different SDL places (MessageHelper, RAI request, Application Manager etc.). This conversion functions are duplicates of each other in terms of conversion from string to enum and vice versa. That's why there is a possibility to replace this functions/methods with unified template functions, located in MessageHelper, and delete redundant code or mark it as deprecated (in case public methods).

List of affected functions:
- AppHMITypeToString/StringToAppHMIType
- HMIResultFromString/HMIResultToString
- MobileResultFromString/MobileResultToString
- StringToHMILevel/StringifiedHMILevel
- MobileLanguageToString/MobileLanguageFromString
- CommonLanguageFromString/CommonLanguageToString
- CommonLightNameFromString

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ychernysheva** [comment]
@theresalech This PR is ready for review.
Thank you!

**Comment by theresalech** [comment]
Hi @ychernysheva, issue 3448 is not currently planned for the Core 6.2 release, but we will work to include if time allows.

**Comment by jacobkeeler** [comment]
@ychernysheva also fix merge conflicts

**Comment by ychernysheva** [comment]
@jacobkeeler All conflicts were resolved in 460daf4, deprecated methods were deleted in 3ae23f2, also actualization after merge of current develop was made in 663933d.

But due to unclear regression status on current develop branch (too many fails on 4c9f873) we cannot define regression status on branch with this fix. May I suggest to wait until develop becomes more stable to merge it and then check regression status?

File: src/components/application_manager/test/hmi_capabilities.json
Comment (jacobkeeler) [inline src/components/application_manager/test/hmi_capabilities.json:1]: Why was this file added and why wasn't it needed before?

**Comment by jacobkeeler** [comment]
@ychernysheva unfortunately there seem to be new merge conflicts and there is a cppcheck issue with `hmi_capabilities_test.cc`:
```
$ cppcheck --force -isrc/3rd_party -isrc/3rd_party-static --quiet --error-exitcode=1 src
[src/components/application_manager/test/hmi_capabilities_test.cc:1488]: (error) Invalid number of character '{' when these macros are defined: ''.
```

File: src/components/application_manager/test/hmi_capabilities.json
Comment (ychernysheva) [reply src/components/application_manager/test/hmi_capabilities.json]: This file was added by mistake during resolving of merge conflicts. Now this file is absent on develop branch, so we don't need it. Deleted in d663395

**Comment by ychernysheva** [comment]
@jacobkeeler Merge conflicts were resolved, also suggested change was implemented in 430a53f.
Full regression check is passed successfully.

**Comment by jacobkeeler** [comment]
@LitvinenkoIra a recent change added another place where this will need to be done: https://github.com/smartdevicelink/sdl_core/blob/release/7.0.0/src/components/application_manager/src/resumption/resumption_data_processor_impl.cc#L52

**Comment by ychernysheva** [comment]
@jacobkeeler Proposed change is implemented in 78a5677

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**
